src/ServiceServerBase.hpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * This software is distributed under the terms of the MIT License.
+ * Copyright (c) 2020-2023 LXRobotics.
+ * Author: Alexander Entinger <alexander.entinger@lxrobotics.com>
+ * Contributors: https://github.com/107-systems/107-Arduino-Cyphal/graphs/contributors.
+ */
+
+#ifndef INC_107_ARDUINO_CYPHAL_SERVICE_SERVER_BASE_HPP
+#define INC_107_ARDUINO_CYPHAL_SERVICE_SERVER_BASE_HPP
+
+/**************************************************************************************
+ * INCLUDE
+ **************************************************************************************/
+
+#include <memory>
+
+#include "SubscriptionBase.h"
+
+/**************************************************************************************
+ * NAMESPACE
+ **************************************************************************************/
+
+namespace impl
+{
+
+/**************************************************************************************
+ * CLASS DECLARATION
+ **************************************************************************************/
+
+class ServiceServerBase : public SubscriptionBase
```

Comment:
Same question here -- does it have to be public?

Reply:
If the inheritance is private I won't be able to access `canard_rx_subscription` as required [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/fix-176/src/Node.ipp#L48).